modules/src/application/ics20_fungible_token_transfer/msgs/transfer.rs

```diff
@@ -0,0 +1,89 @@
+use std::convert::{TryFrom, TryInto};
+
+use tendermint::account::Id as AccountId;
+use tendermint_proto::Protobuf;
+
+use ibc_proto::ibc::applications::transfer::v1::MsgTransfer as RawMsgTransfer;
+
+use crate::address::{account_to_string, string_to_account};
+use crate::application::ics20_fungible_token_transfer::error::{Error, Kind};
+use crate::ics02_client::height::Height;
+use crate::ics24_host::identifier::{ChannelId, PortId};
+use crate::tx_msg::Msg;
+
+pub const TYPE_URL: &str = "/ibc.applications.transfer.v1.MsgTransfer";
+
+///
+/// Message definition for the "packet receiving" datagram.
+///
+#[derive(Clone, Debug, PartialEq)]
+pub struct MsgTransfer {
+    /// the port on which the packet will be sent
+    pub source_port: PortId,
+    /// the channel by which the packet will be sent
+    pub source_channel: ChannelId,
+    /// the tokens to be transferred
+    pub token: Option<ibc_proto::cosmos::base::v1beta1::Coin>,
```

Comment:
At some point (perhaps in a later PR), we should define this `Coin` type in a module rather than rely on the protobuf definition directly.